### PR TITLE
[BH-1236] Add temperature compensation

### PR DESCRIPTION
**Pull request by tigransoghbatyan-digica**
Add temperature internal heating compensation

### CODE REVIEW

File: products/BellHybrid/services/evtmgr/EventManager.cpp [hunk 1]
```diff
@@ -119,6 +108,16 @@ void EventManager::initProductEvents()
...
+
+    connect(service::AudioPlaybackStartNotification(), [&](sys::Message *msgl) {
+        temperatureProvider.audioStarted();
+        return sys::msgHandled();
+    });
+
+    connect(service::AudioPlaybackStopNotification(), [&](sys::Message *msgl) {
+        temperatureProvider.audioStoped();
+        return sys::msgHandled();
+    });
```
Comment (wojtekrzepecki) [inline products/BellHybrid/services/evtmgr/EventManager.cpp:120]: In my opinion dependency to ServiceAudio is wrong here. More general solution will be to use our PowerManager to obtain current cpu load and cpu frequency to provide useful input to temperature compensation. Was the concept for this audio playback compensation discussed before in Harmony team?

Comment (swichu91) [reply products/BellHybrid/services/evtmgr/EventManager.cpp:120]: I agree with Wojtek. The same goes for the audio service which now requires event manager dependency. As far as I know, it was found audio service is the main source of the temperature rise. Of course, the solution proposed by Wojtek is much better.

File: products/BellHybrid/services/audio/ServiceAudio.cpp [hunk 2]
```diff
@@ -171,7 +173,7 @@ namespace service
         if (it != retCodes.end()) {
             return std::make_unique<AudioStopResponse>(it->second, it->first);
         }
-
+        bus.sendUnicast(std::make_shared<service::AudioPlaybackStopNotification>(), service::name::evt_manager);
```
Comment (alekrudnik) [inline products/BellHybrid/services/audio/ServiceAudio.cpp:176]: why do you send it explicitly to event manager?

Comment (tigransoghbatyan-digica) [reply products/BellHybrid/services/audio/ServiceAudio.cpp:176]: according to current implementation, the temperature compensation algorithm is implemented in EventManager Service

File: products/BellHybrid/services/evtmgr/temperature-compensation/TemperatureProvider.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+{
+    auto diff =
+        std::chrono::duration_cast<std::chrono::seconds>(std::chrono::system_clock::now() - audioLastTimeStateChange);
+    auto secondsElapsed = diff.count();
+    auto decay          = std::exp(-secondsElapsed / audioCompensationTimeCoefficient);
+    if (audioPlaying) {
+        audioCompensationValue += (audioCompensationAmplitude - audioCompensationValue) * (1 - decay);
+    }
+    else {
+        audioCompensationValue = (audioCompensationValue)*decay;
```
Comment (mjan-dev) [inline products/BellHybrid/services/evtmgr/temperature-compensation/TemperatureProvider.cpp:48]:
```suggestion
        audioCompensationValue *= decay;
```

**Comment by mjan-dev** [comment]
Closing as this solution was tested and seems to be not effective.

**Pull request closed without merging**